test/jdk/java/awt/Frame/Iconify/IconifyTest.java

```diff
@@ -0,0 +1,188 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8296934
+ * @summary Verifies whether Undecorated Frame can be iconified or not.
+ * @run main IconifyTest
+ */
+public class IconifyTest {
+
+    private static Robot robot;
+    private static Button button;
+    private static Frame frame;
+    private static volatile int windowStatusEventType;
+    private static volatile int windowIconifiedEventType;
+    private static volatile boolean focusGained = false;
+
+    public static void initializeGUI() {
+        frame = new Frame();
+        frame.setLayout(new FlowLayout());
+        frame.setSize(200, 200);
+        frame.setUndecorated(true);
+
+        frame.addWindowFocusListener(new WindowAdapter() {
+            public void windowGainedFocus(WindowEvent event) {
+                focusGained = true;
+            }
+        });
+
+        frame.addWindowListener(new WindowAdapter() {
+            public void windowActivated(WindowEvent e) {
+                windowStatusEventType = WindowEvent.WINDOW_ACTIVATED;
+                System.out.println("Event encountered: " + e);
+            }
+
+            public void windowIconified(WindowEvent e) {
+                windowIconifiedEventType = WindowEvent.WINDOW_ICONIFIED;
+                System.out.println("Event encountered: " + e);
+            }
+
+            public void windowDeiconified(WindowEvent e) {
+                windowIconifiedEventType = WindowEvent.WINDOW_DEICONIFIED;
+                System.out.println("Event encountered: " + e);
+            }
+
+            public void windowDeactivated(WindowEvent e) {
+                windowStatusEventType = WindowEvent.WINDOW_DEACTIVATED;
+                System.out.println("Event encountered: " + e);
+            }
+        });
+
+        button = new Button("Minimize me");
+        button.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                frame.setExtendedState(Frame.ICONIFIED);
+            }
+        });
+
+        frame.setBackground(Color.green);
+        frame.add(button);
+        frame.setLocationRelativeTo(null);
+        frame.toFront();
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws AWTException, InvocationTargetException, InterruptedException {
```

Comment:
Please split the long lines to have 80 chars per line.

Reply:
Done the changes